### PR TITLE
Database tuning

### PR DESCRIPTION
**Pull request by berlinguyinca**
We are seeing several performance slowdowns and losses, due to the system not processing trades in time. During the investigation, I noticed that the Trade table is not containing an index on a system, which should be index.

This causes slowdown over time and seq scans when index scans should be used.

During a test, of evaluating several different strategies, one set with index and one set without an index. It can be clearly seen that this causes unnecessary losses.

Example results of a 24h test, all run on the same system, same docker version, started at the same time (+/-5seconds) this behavior is clear demonstrated.

With Index
```

╒════════════════════════════════════════════════════════════╤═════════════════╤═════════════════╕
│ database                                                   │   percent_stake │   percent_total │
╞════════════════════════════════════════════════════════════╪═════════════════╪═════════════════╡
│ bots/data/with-index/trades.MultiRSI.sqlite                │       -8.99108  │     -0.0449554  │
├────────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/with-index/trades.MultiRSI2.sqlite               │      -36.2941   │     -0.18147    │
├────────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/with-index/trades.MultiRSICMF.sqlite             │      -10.6663   │     -0.0533313  │
├────────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/with-index/trades.MultiRSIMVA.sqlite             │      -21.9627   │     -0.109814   │
├────────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/with-index/trades.MultiRSITunedBuyAndSell.sqlite │        4.03005  │      0.0201503  │
├────────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/with-index/trades.MultiRSITunedSell.sqlite       │        3.25623  │      0.0162812  │
├────────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/with-index/trades.MultiTest.sqlite               │       10.4401   │      0.0522004  │
├────────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/with-index/trades.MultiTest2.sqlite              │       16.5727   │      0.0828636  │
├────────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/with-index/trades.MultiTest3.sqlite              │        2.97717  │      0.0148859  │
├────────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/with-index/trades.MultiTest4.sqlite              │       55.8319   │      0.27916    │
├────────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/with-index/trades.MultiTest7.sqlite              │        0.438596 │      0.00219298 │
├────────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/with-index/trades.MultiTest8.sqlite              │       35.2801   │      0.176401   │
├────────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/with-index/trades.Scalp2.sqlite                  │       13.4801   │      0.0674005  │
├────────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/with-index/trades.TheBigScalper.sqlite           │      -34.4922   │     -0.172461   │
├────────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/with-index/trades.TrendTest1.sqlite              │       -0.221607 │     -0.00110803 │
╘════════════════════════════════════════════════════════════╧═════════════════╧═════════════════╛

```  

No index:
```
╒══════════════════════════════════════════════════════════╤═════════════════╤═════════════════╕
│ database                                                 │   percent_stake │   percent_total │
╞══════════════════════════════════════════════════════════╪═════════════════╪═════════════════╡
│ bots/data/no-index/trades.MultiRSI.sqlite                │      -10.9875   │     -0.0549373  │
├──────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/no-index/trades.MultiRSI2.sqlite               │      -35.9454   │     -0.179727   │
├──────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/no-index/trades.MultiRSICMF.sqlite             │       -6.54824  │     -0.0327412  │
├──────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/no-index/trades.MultiRSIMVA.sqlite             │      -21.2606   │     -0.106303   │
├──────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/no-index/trades.MultiRSITunedBuyAndSell.sqlite │        4.03005  │      0.0201503  │
├──────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/no-index/trades.MultiRSITunedSell.sqlite       │        2.65381  │      0.0132691  │
├──────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/no-index/trades.MultiTest.sqlite               │       10.002    │      0.05001    │
├──────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/no-index/trades.MultiTest2.sqlite              │       16.696    │      0.0834798  │
├──────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/no-index/trades.MultiTest3.sqlite              │       -3.01467  │     -0.0150734  │
├──────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/no-index/trades.MultiTest4.sqlite              │       29.2337   │      0.146169   │
├──────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/no-index/trades.MultiTest7.sqlite              │        0.438596 │      0.00219298 │
├──────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/no-index/trades.MultiTest8.sqlite              │       20.9279   │      0.104639   │
├──────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/no-index/trades.Scalp2.sqlite                  │       14.6038   │      0.0730191  │
├──────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/no-index/trades.TheBigScalper.sqlite           │      -34.6553   │     -0.173276   │
├──────────────────────────────────────────────────────────┼─────────────────┼─────────────────┤
│ bots/data/no-index/trades.TrendTest1.sqlite              │       -0.221607 │     -0.00110803 │
╘══════════════════════════════════════════════════════════╧═════════════════╧═════════════════╛
```

Losses are consistently higher without using an index.

This PR also fixes some dependency issues I saw, but if this should be left out, just merge:

85c73ea8507d012353bb9744b4371d978bf07af6

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,6 +13,11 @@ ENV LD_LIBRARY_PATH /usr/local/lib
 RUN mkdir /freqtrade
 WORKDIR /freqtrade
 
+# Update PIP
+RUN python -m pip install --upgrade pip
+RUN pip install future
+RUN pip install numpy
```
Comment (xmatthias) [inline Dockerfile]: what exactly is future for ... ?
not saying it's not needed, just trying to understand

also - i would try to combine multiple consecutive `RUN` statements to one (using `\` and `&&`, as this keeps the image iterations lower, and therefore also the size.

and numpy is there, line 23 ... :)

File: requirements.txt [hunk 2]
```diff
@@ -23,3 +23,13 @@ scikit-optimize==0.5.2
...
 #plotly==3.0.0
+
+# Required for plotting data
+plotly==3.0.0
+
+# find first, C search in arrays
+py_find_1st==1.1.1
+
+#Load ticker files 30% faster
+ujson==1.35
```
Comment (xmatthias) [inline requirements.txt]: not sure about this files changes

ujson has it's own PR - and is not used by develop atm, so i would not force it here.
Also, py_find_1st should be included in the PR requiring it (which i suspect will be backslash?)

plotly on the other hand is still there as optional dependency - not needed in a docker-container at all...

**Comment by xmatthias** [comment]
i am hesitant to only merge one commit, as part of the 2nd commit does make sense too (git as dependency, upgrade pi).

one question, if i have an existing DB - will that be upgraded too or does the index only apply to newly created databases?

other than that, good job :+1:

**Comment by berlinguyinca** [comment]
the index will only apply to new db's. Don't know how to update old ones
vendors independently.

no idea what futures is needed for, but since recent develop it failed
without it. I think it might be related to some dependency update. Most
python installations have it installed by default, it basically makes
python2 code mostly python3 compliant.

**Comment by berlinguyinca** [comment]
regarding plotly, some people have strategies which plot stuff during
execution in docker.

agree with the other dependencies and I update the pr tomorrow to remove
them

**Comment by berlinguyinca** [comment]
won't work with requirements or i would have done it. has to be done
manually.

**Comment by xmatthias** [comment]
from pypi https://pypi.org/project/future/
> The imports have no effect on Python 3

i don't understand why we need this installed on a project supporting >= python3.6 only ...

**Comment by berlinguyinca** [comment]
neither do i, i just added it to make things work.

**Comment by berlinguyinca** [comment]
the build just doesn't work without it. feel free to drop it and just merge
the db change as suggested.

**Comment by xmatthias** [comment]
No reason to feel offended, i was asking "what is the problem you're seeing" because i don't see any problem with the latest Docker file, but i would like to understand the problem, to potentially find a fix.

**Comment by xmatthias** [comment]
i've reverted the dependency issues for now to get the DB changes in as i think they'll benefit general performance..

But still would like to understand the dependency issue you saw.
Are the issues related to `technical` - or running the bot in Docker generally (also without technical)

**Pull request merged**